db/upgrade.php

```diff
@@ -154,5 +154,15 @@ function xmldb_tool_objectfs_upgrade($oldversion) {
 
         upgrade_plugin_savepoint(true, 2021122306, 'tool', 'objectfs');
     }
+
+    if ($oldversion < 2021122307) {
+        // Check to make sure adhoc task not already running.
+        if (!$DB->record_exists('task_scheduled', ['classname' => '\tool_objectfs\task\populate_objects_filesize'])) {
```

Comment:
this is the wrong table, should be task_adhoc

Reply:
good catch